privacyidea/lib/authcache.py

```diff
@@ -89,7 +95,8 @@ def cleanup(minutes):
     return r
 
 
-def verify_in_cache(username, realm, resolver, password, first_auth=None, last_auth=None):
+def verify_in_cache(username, realm, resolver, password, first_auth=None, last_auth=None,
+                    max_number_of_authentications=None):
```

Comment:
since this is basically an integer value, the default should be `0`

Reply:
Should values <= 0 be ignored then, or should an exception be thrown? See https://github.com/privacyidea/privacyidea/pull/2635/commits/6a9199e1db3b4f931e7c070f2ae0c305ae782bb7

Reply:
negative values don't make sense here so if someone passes a negative value for the number of auths the cache should not be used (in my opinion). So the test if the value is greater than 0 is enough.